AssertionError: Type info for django.db.models.query._QuerySet missing

# VScode mypy Django QuerySet error

Hi, I am not using anything out of the ordinary other than mypy and django-stubs. When my VScode runs mypy it crashes with this traceback below.


```
  File "/venv/lib/python3.11/site-packages/mypy_django_plugin/main.py", line 50, in transform_model_class
    process_model_class(ctx, django_context)
  File "/venv/lib/python3.11/site-packages/mypy_django_plugin/transformers/models.py", line 674, in process_model_class
    initializer_cls(ctx, django_context).run()
  File "/venv/lib/python3.11/site-packages/mypy_django_plugin/transformers/models.py", line 96, in run
    self.run_with_model_cls(model_cls)
  File "/venv/lib/python3.11/site-packages/mypy_django_plugin/transformers/models.py", line 377, in run_with_model_cls
    fallback_manager_info = self.get_or_create_manager_with_any_fallback()
                            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/venv/lib/python3.11/site-packages/mypy_django_plugin/transformers/models.py", line 131, in get_or_create_manager_with_any_fallback
    fallback_queryset = self.get_or_create_queryset_with_any_fallback()
                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/venv/lib/python3.11/site-packages/mypy_django_plugin/transformers/models.py", line 181, in get_or_create_queryset_with_any_fallback
    assert base_queryset_info, f"Type info for {fullnames.QUERYSET_CLASS_FULLNAME} missing"
AssertionError: Type info for django.db.models.query._QuerySet missing
```


mypy almost always seems to crash with VScode open. I had originally patched files here

```
File "/venv/lib/python3.11/site-packages/mypy_django_plugin/transformers/models.py", line 155, in get_or_create_queryset_with_any_fallback
```

But it was a very crude way of getting it to work.

It does on occasion work for a few minutes (or at least not error) but I cannot seem to replicate its success on command.

I have been through several open and closed issues closely related but can't find anything similar to what I am facing.

I also don't know if this bug will be covered by this release. - https://github.com/typeddjango/django-stubs/issues/1316

Would be grateful for any help on this as its stopping my VScode from running mypy on my code and get a popup appear everytime I do a code change. 


## System information

- OS: Mac M2 Ventura 13.1
- `python` version: 3.11.0
- `django` version: 4.1.4
- `mypy` version: 0.991
- `django-stubs` version: 1.13.0
- `django-stubs-ext` version: 0.7.0
- `vscode` version: 1.74.3
- 'vscode-mypy-extension` version: v0.2.2

--- a/mypy_django_plugin/transformers/models.py
+++ b/mypy_django_plugin/transformers/models.py
@@ -103,7 +103,7 @@ class ModelClassInitializer:
         # Not a generated manager
         return None
 
-    def get_or_create_manager_with_any_fallback(self, related_manager: bool = False) -> TypeInfo:
+    def get_or_create_manager_with_any_fallback(self, related_manager: bool = False) -> Optional[TypeInfo]:
         """
         Create a Manager subclass with fallback to Any for unknown attributes
         and methods. This is used for unresolved managers, where we don't know
@@ -121,11 +121,14 @@ class ModelClassInitializer:
             return manager_info
 
         fallback_queryset = self.get_or_create_queryset_with_any_fallback()
+        if fallback_queryset is None:
+            return None
         base_manager_fullname = (
             fullnames.MANAGER_CLASS_FULLNAME if not related_manager else fullnames.RELATED_MANAGER_CLASS
         )
         base_manager_info = self.lookup_typeinfo(base_manager_fullname)
-        assert base_manager_info, f"Type info for {base_manager_fullname} missing"
+        if base_manager_info is None:
+            return None
 
         base_manager = fill_typevars(base_manager_info)
         assert isinstance(base_manager, Instance)
@@ -153,7 +156,7 @@ class ModelClassInitializer:
 
         return manager_info
 
-    def get_or_create_queryset_with_any_fallback(self) -> TypeInfo:
+    def get_or_create_queryset_with_any_fallback(self) -> Optional[TypeInfo]:
         """
         Create a QuerySet subclass with fallback to Any for unknown attributes
         and methods. This is used for the manager returned by the method above.
@@ -168,7 +171,8 @@ class ModelClassInitializer:
             return queryset_info
 
         base_queryset_info = self.lookup_typeinfo(fullnames.QUERYSET_CLASS_FULLNAME)
-        assert base_queryset_info, f"Type info for {fullnames.QUERYSET_CLASS_FULLNAME} missing"
+        if base_queryset_info is None:
+            return None
 
         base_queryset = fill_typevars(base_queryset_info)
         assert isinstance(base_queryset, Instance)
@@ -346,11 +350,12 @@ class AddManagers(ModelClassInitializer):
                 # ignoring a more specialised manager not being resolved while still
                 # setting _some_ type
                 fallback_manager_info = self.get_or_create_manager_with_any_fallback()
-                self.add_new_node_to_model_class(
-                    manager_name,
-                    Instance(fallback_manager_info, [Instance(self.model_classdef.info, [])]),
-                    is_classvar=True,
-                )
+                if fallback_manager_info is not None:
+                    self.add_new_node_to_model_class(
+                        manager_name,
+                        Instance(fallback_manager_info, [Instance(self.model_classdef.info, [])]),
+                        is_classvar=True,
+                    )
 
                 # Find expression for e.g. `objects = SomeManager()`
                 manager_expr = self.get_manager_expression(manager_name)
@@ -493,9 +498,10 @@ class AddRelatedManagers(ModelClassInitializer):
                     # See https://github.com/typeddjango/django-stubs/pull/993
                     # for more information on when this error can occur.
                     fallback_manager = self.get_or_create_manager_with_any_fallback(related_manager=True)
-                    self.add_new_node_to_model_class(
-                        attname, Instance(fallback_manager, [Instance(related_model_info, [])])
-                    )
+                    if fallback_manager is not None:
+                        self.add_new_node_to_model_class(
+                            attname, Instance(fallback_manager, [Instance(related_model_info, [])])
+                        )
                     related_model_fullname = related_model_cls.__module__ + "." + related_model_cls.__name__
                     self.ctx.api.fail(
                         (

